lib/Backend/IRBuilder.cpp

```diff
@@ -1677,8 +1677,8 @@ IRBuilder::BuildReg1(Js::OpCode newOpcode, uint32 offset, Js::RegSlot R0)
     case Js::OpCode::Throw:
         {
             srcOpnd = this->BuildSrcOpnd(srcRegOpnd);
-
-            if (this->catchOffsetStack && !this->catchOffsetStack->Empty())
+            if ((this->handlerOffsetStack && !this->handlerOffsetStack->Empty()) ||
+                inFinallyBlock > 0)
```

Comment:
>inFinallyBlock [](start = 16, length = 14)

nit: for better readability, change name to finallyBlockLevel or something or have an API InFinallyBlock.